### DISCUSSION TITLE
Securing access to your Cloudflare account

### DISCUSSION THREAD

**Discussion by IrmaIsfot** [homeassistant-apps/app-cloudflared#542] [Q&A]
I've read the paragraph [Securing access to your Cloudflare account](https://github.com/brenner-tobias/addon-cloudflared/wiki/How-tos#securing-access-to-your-cloudflare-account) in the How-tos.

First of all I wonder: how could or why would my cert.pem become compromised?
Is this something I need to worry about? 
How do I prevent it?
And perhaps most importantly: **how would I know** if it has become compromised?
Obviously, I'm a total newbie at this level of networking...

So, if the cert.pem is at risk for me:
I have created a secondary account and invited that to my main account, so when I login to the secondary account, I get to chose wich account to view:
![secondary1](https://github.com/brenner-tobias/addon-cloudflared/assets/20162763/7b389948-fc8c-461f-81d1-0ec194acf29f)
And then, I get to see the primary account dashboard, with my domain connected to it:
![secondary2](https://github.com/brenner-tobias/addon-cloudflared/assets/20162763/07b0c604-83fb-45e8-9d5a-faa7f807b6de)

But then..?
"2. Instead of using your primary account to authenticate the tunnel, use your secondary account."

I installed the Cloudflared add-on and set my domain in the configuration and then started the add-on.
Followed the link in the log, but that led me to my primary account. I've now realized that happened since I was already logged in to my primary account at that time.
But since I did **not** click on the blue Authorize button, I didn't mess up and received the cert for my primary account, right?

So, instead I have now logged in with my secondary account and selected the dashboard of my primary.
Now, using the link from the log, I see the primary account tunnel, from my secondary account.
![secondary4](https://github.com/brenner-tobias/addon-cloudflared/assets/20162763/21d3cdef-9b82-41c4-b7a2-8c23b87d5126)

So if I now click the blue Authorize button, I will receive the secondary account cert file?
![secondary5](https://github.com/brenner-tobias/addon-cloudflared/assets/20162763/b06c4f19-36e0-485f-bc68-a58176528402)

Then, I can continue with the rest of the configuration...

And if the cert do become compromised and I am able to find that out, I would revoke my secondary account.
![secondary6](https://github.com/brenner-tobias/addon-cloudflared/assets/20162763/9928042f-b2d4-4277-ace1-3d5482b2d544)

This would render my cert invalid?  So the Cloudflared plug-in would no longer be able to use the tunnel?
But I'd still have my domain connected to my primary, uncompromised account.
I would then have to create and invite a third account and create a new tunnel, authorized from the third account?

I think I've understood this process now...

But if I had **only** used my first, primary account, and later found out the cert was compromised, 
why not just disconnect my domain and abandon that primary account if that happens, 
and then create a new account at that point..?
Is the compromised cert valid for any future tunnels associated with my domain?

Sorry for the long post! There is so much to learn, and I really want to understand what I am doing...

**Comment by brenner-tobias** [comment]
Thanks for raising this. You are right that this is somewhat complicated. The reason why this might be a problem that it is (or maybe was, I did just ask in [this old issue](https://github.com/cloudflare/cloudflared/issues/93)) not possible to revoke the cert.pem access. The key file is nothing more then a text-document that is stored wherever you store it. In our case, it is stored in the add-on file-space. 

Now if someone manages to get access to your HA installation and also access to all files, he will be able to simply copy this certificate file and use it for whatever he wants. You will not be able to detect that unless you see suspicious activity in your Cloudflare Account.

If you do, you are completely right to delete the account that was compromised. Since it is a hassle to re-assign the domain to a new Cloudflare account, we recommend to create a secondary one for this, but if you did not do it, it is also possible to simply move the domain to a new Cloudflare account.

I hope this helps, let me know if there was anything unclear.
